Add open document option

jeffa5 opened the pull request:
Fixes https://github.com/tectonic-typesetting/tectonic/issues/109

Adds the `--open` option to the `build` subcommand of the new v2 cli.

pkgw commented on src/document.rs:
```diff
@@ -372,6 +380,24 @@ impl Document {
+                status.report(
+                    MessageKind::Error,
+                    format_args!(
+                        "Failed to open `{}` with system handler",
```
Here too, the note text should start with a lower-case letter.

Is it possible to use:

```
if let Err(e) = open::that(&out_file) {
   tt_error!("... text ....", out_file.display(); e);
}
```

?

jeffa5 replied:
Yep, that looks better, changed.

pkgw reviewed (commented):
Thanks so much! This looks great and I just have a few comments.

1. It would be great to get some test coverage, although I can see how that might be tricky with the behavior that's being added here. Maybe the `open` crate has some test infrastructure that we can copy?
2. It would also be great to add the new option to the docs in `docs/src/v2cli/build.md`

Below I've also made a few very small comments.

jeffa5 commented on Cargo.toml:
```diff
@@ -84,6 +84,7 @@ tectonic_xdv = { path = "crates/xdv", version = "0.0.0-dev.0" }
 termcolor = "^1.1"
 toml = { version = "^0.5", optional = true }
 zip = { version = "^0.5", default-features = false, features = ["deflate"] }
+open = "1.4.0"
```
Fixed, also sorted the other dependencies.

jeffa5 commented on src/document.rs:
```diff
@@ -372,6 +380,24 @@ impl Document {
+                    .with_extension(match profile.target_type {
+                        BuildTargetType::Pdf => "pdf",
+                    });
+            status.note_highlighted("Opening ", &format!("`{}`", out_file.display()), "");
```
I've changed this to use the `tt_note` macro now too.

jeffa5 commented:
I've added the docs and addressed the comments.

Regarding testing I've added one with the `BROWSER` environment variable set. This seems to work with not opening the document and passes the test but the `open` crate doesn't seem to have any tests either.

pkgw commented:
Brilliant!

Now that I have [awesome release automation](https://pkgw.github.io/cranko/) I try to get contributions into releases quickly, but FYI it will take a little while as I work through the infrastructural changes that I've been pursuing.
